### ISSUE TITLE
How to provide CanonicalUser Principal in the new version?

### ISSUE THREAD

**Issue by pbudzon** [cloudtools/awacs#80]
It seems with the latest release, it is no longer possible to specify `CanonicalUser` as a principal? This type of principal is used for S3 BucketPolicy, with Statement:
```
 aws.Statement(
                        Effect=aws.Allow,
                        Principal=aws.Principal(
                            "CanonicalUser", "some canonical id"
                        ),
                        Action=[
                            aws.Action("s3", "GetObject"),
                        ],
                        Resource=[
                            Join("", ["arn:aws:s3:::", Ref(self.bucket_name), "/*"]),
                        ],
                    ),
```

the above now fails with:
```
ValueError: Principal must be one of: AWS, Federated, Service
```

**Comment by markpeek** [comment]
Thanks for the issue and sorry for the regression. We'll see about spinning a release later to include this change.

**Comment by austinried** [comment]
So what was the resolution on this, why is it closed? Was there code merged somewhere that fixes the issue? As far as I can tell this is still a problem with the most recent version, which is the version where the problem appeared, 0.7.0.

**Comment by markpeek** [comment]
@austinried my apologies, it was closed due to commit 2193fae12b69245a8e033de1e8d627d223bb6dd1 (shown above) but we neglected to issue a new release. This likely will not happen for a couple of days until I get back to my signing keys unless @phobologic could spin one sooner.

**Comment by austinried** [comment]
Ah I complete missed that commit, thanks for the update!
